crates/polars-ops/src/series/ops/ewm_by.rs

```diff
@@ -0,0 +1,173 @@
+        },
+        #[cfg(feature = "dtype-datetime")]
+        (_, DataType::Datetime(time_unit, _)) => {
+            let half_life = adjust_half_life_to_time_unit(half_life, time_unit);
+            ewm_mean_by(s, &times.cast(&DataType::Int64)?, half_life, assume_sorted)
+        },
+        #[cfg(feature = "dtype-date")]
+        (_, DataType::Date) => ewm_mean_by(
+            s,
+            &times.cast(&DataType::Datetime(TimeUnit::Milliseconds, None))?,
+            half_life,
+            assume_sorted,
+        ),
+        (_, DataType::UInt64 | DataType::UInt32 | DataType::Int32) => {
+            ewm_mean_by(s, &times.cast(&DataType::Int64)?, half_life, assume_sorted)
+        },
+        (DataType::UInt64 | DataType::UInt32 | DataType::Int64 | DataType::Int32, _) => {
+            ewm_mean_by(
+                &s.cast(&DataType::Float64)?,
+                times,
+                half_life,
+                assume_sorted,
+            )
+        },
+        _ => {
+            polars_bail!(InvalidOperation: "expected series to be Float64, Float32, \
+                Int64, Int32, UInt64, UInt32, and `by` to be Date, Datetime, Int64, Int32, \
+                UInt64, or UInt32")
+        },
+    }
+}
+
+/// Sort on behalf of user
+fn ewm_mean_by_impl<T>(
+    values: &ChunkedArray<T>,
+    times: &Int64Chunked,
+    half_life: i64,
+) -> ChunkedArray<T>
+where
+    T: PolarsFloatType,
+    T::Native: Float + Zero + One,
+    ChunkedArray<T>: ChunkTakeUnchecked<IdxCa>,
+{
+    let sorting_indices = times.arg_sort(Default::default());
+    let values = unsafe { values.take_unchecked(&sorting_indices) };
+    let times = unsafe { times.take_unchecked(&sorting_indices) };
+
+    let mut out = vec![None; times.len()];
+
+    let mut skip_rows: usize = 0;
+    let mut prev_time: i64 = 0;
+    let mut prev_result = T::Native::zero();
+    for (idx, (value, time)) in values.iter().zip(times.iter()).enumerate() {
+        if let (Some(time), Some(value)) = (time, value) {
+            prev_time = time;
+            prev_result = value;
+            unsafe {
+                let out_idx = sorting_indices.get_unchecked(idx).unwrap();
+                *out.get_unchecked_mut(out_idx as usize) = Some(prev_result);
+            }
+            skip_rows = idx + 1;
+            break;
+        };
+    }
+    values
+        .iter()
+        .zip(times.iter())
+        .enumerate()
+        .skip(skip_rows)
+        .for_each(|(idx, (value, time))| {
+            let result_opt = match (time, value) {
+                (Some(time), Some(value)) => {
+                    let result = update(value, prev_result, time, prev_time, half_life);
+                    prev_time = time;
+                    prev_result = result;
+                    Some(result)
+                },
+                _ => None,
+            };
+            unsafe {
+                let out_idx = sorting_indices.get_unchecked(idx).unwrap();
```

Comment:
Instead of going through all buffers of `sorting_idx` that is `ChunkedArray`. We should use `cont_slice` to get access to the slice beneath. That saves a lot of accumulated indirections.